Source/WebCore/rendering/style/StyleImage.h

```diff
@@ -74,6 +74,7 @@ class StyleImage : public RefCounted<StyleImage> {
     virtual bool imageHasRelativeWidth() const = 0;
     virtual bool imageHasRelativeHeight() const = 0;
     virtual float imageScaleFactor() const { return 1; }
+    virtual bool imageHasNaturalDimensions() const { return true; }
```

Comment:
I think you probably want to implement this for StyleMultiImage call through to the selected image. Would be a good test case well (gradient in an image-set()).

Any reason to keep `usesImageContainerSize()` around?